components/zigbee/zigbee_attribute.h

```diff
@@ -62,16 +65,27 @@ class ZigBeeAttribute : public Component {
   bool set_attr_requested_{false};
 };
 
-template<typename T> void ZigBeeAttribute::add_attr(uint8_t attr_access, T value_p) {
+template<typename T> void ZigBeeAttribute::add_attr(uint8_t attr_access, uint8_t max_size, T value_p) {
   this->zb_->add_attr(this, this->endpoint_id_, this->cluster_id_, this->role_, this->attr_id_, this->attr_type_,
-                      attr_access, value_p);
+                      attr_access, max_size, value_p);
 }
 
 template<typename T> void ZigBeeAttribute::set_attr(T *value_p) {
```

Comment:
I guess we could also change that to `set_attr(T value_p)` and move conversions here if that makes it easier. Now it is a bit confusing, numeric types require to convert everything and to create the variable with `new` in the calling function, but for strings/char[] the variable is created here.

Reply:
Giving that another thought, I would say that is the way to go. It requires a larger refactor, though. Feel free to ignore this comment (and the first one), we can merge this first and then I can work on the refactor later.